### PR TITLE
feat(System): Labs Alpha

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
- Creates a basic theme and theme provider 
- Creates initial style functions
- Creates a basic text component
- Renames typography scale to singular scale.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.4.2-alpha.b3e40f.0<br/>@codecademy/gamut-styles@7.1.0-alpha.b3e40f.0<br/>@codecademy/gamut-system@0.0.1-alpha.b3e40f.0<br/>@codecademy/gamut@16.2.4-alpha.b3e40f.0<br/>@codecademy/markdown-overrides@0.3.4-alpha.b3e40f.0<br/>@codecademy/styleguide@17.7.4-alpha.b3e40f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f970c7c8f74a05542c3134e--gamut-preview.netlify.app>https://5f970c7c8f74a05542c3134e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f970c7c8f74a05542c3134e>Deploy Logs</a></p>

File: packages/gamut-system/src/core/system/index.ts [hunk 1]
```diff
@@ -67,7 +67,9 @@ const create = <
 };
 
 export const system = {
-  create,
+  create: <Config extends SystemConfig<{}>>(config: Config) => {
+    return create<{}, Config>(config);
+  },
```
Comment (JoshuaKGoldberg) [inline packages/gamut-system/src/core/system/index.ts:72]: Ha, good find on this one..

File: packages/gamut-styles/src/variables/typography.ts [hunk 2]
```diff
@@ -36,27 +35,23 @@ export const fontDecoration = {
...
-  },
-  heading: {
-    xxl: pxRem(base * 3),
-    xl: pxRem(base * 2.2),
-    lg: pxRem(base * 1.6),
-    md: pxRem(base * 1.4),
-    sm: pxRem(base * 1.25),
-    xs: pxRem(base),
-  },
+  ['t-1']: pxRem(64),
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/variables/typography.ts:38]: Can we do `text-1` for clarity? You told us earlier today about this naming system and I still forgot what this meant just looking at it 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
Let's try it out!!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.4.2-alpha.553494.0<br/>@codecademy/gamut-styles@7.1.0-alpha.553494.0<br/>@codecademy/gamut-system@0.0.1-alpha.553494.0<br/>@codecademy/gamut@16.2.4-alpha.553494.0<br/>@codecademy/markdown-overrides@0.3.4-alpha.553494.0<br/>@codecademy/styleguide@17.7.4-alpha.553494.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f973e21a26341190a183450--gamut-preview.netlify.app>https://5f973e21a26341190a183450--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f973e21a26341190a183450>Deploy Logs</a></p>

**Pull request merged**